emmet-core/emmet/core/band_theory.py

```diff
@@ -163,91 +205,134 @@ def to_pmg(
             },
             coords_are_cartesian=False,
             structure=self.structure,
-            projections=projections,
+            projections=self.projections.to_pmg_like() if self.projections else None,
         )
 
 
-class BaseElectronicDos(BandTheoryBase):
-    """Basic structure for spin-resolved density of states (DOS).
+class ProjectedDos(BaseModel):
+    """Atom and orbital projected DOS."""
 
-    Note that this class is intended for storing:
-        1. Total DOS.
-        2. A single-orbital-resolved component of a site- and orbital-
-            projected DOS.
-    """
+    spin_up_densities: list[list[float] | None] | None = Field(
+        None, description="The spin-up projected densities of state."
+    )
 
-    spin_up_densities: list[float] | None = Field(None, description="The spin-up DOS.")
+    spin_down_densities: list[list[float] | None] | None = Field(
+        None, description="The spin-down projected densities of state."
+    )
 
-    spin_down_densities: list[float] | None = Field(
-        None, description="The spin-down DOS."
+    orbital: list[str | None] | None = Field(
+        None, description="The orbital character of this DOS, if applicable."
     )
 
-    energies: list[float] | None = Field(
-        None, description="The energies at which the DOS was calculated."
+    site_index: list[int | None] | None = Field(
+        None,
+        description="The index of the atom in the structure onto which this DOS is projected.",
     )
 
-    efermi: float | None = Field(None, description="The Fermi energy.")
+    def _to_list_of_dict(self) -> list[dict[str, str | int | list[float]]]:
+        """Possible serialization procedure."""
+        return [
+            {k: getattr(self, k)[i] for k in self.__class__.model_fields}
+            for i in range(len(self.spin_up_densities or []))
+        ]
 
-    orbital: str | None = Field(
-        None, description="The orbital character of this DOS, if applicable."
-    )
+    @classmethod
+    def _from_list_of_dict(cls, lod: list[dict[str, str | int | list[float]]]) -> Self:
+        """Possible deserialization route."""
+        return cls(**{k: [entry.get(k) for entry in lod] for k in cls.model_fields})  # type: ignore[arg-type]
```

Comment:
Should the list entry be `None` explicitly when `k not in entry`? I.e.

```python
[entry.get(k) for entry in lod]
```

vs

```python
[entry[k] for entry in lod if k in entry]
```

Reply:
@esoteric-ephemera @tsmathis Thanks for clarifying! Either solution is good I think. Whatever you think is best going forward.